packages/react/src/live-region/Announce.tsx

```diff
@@ -132,12 +132,110 @@ export function Announce({
   )
 }
 
-function getTextContent(element: HTMLElement): string {
-  let value = ''
-  if (element.hasAttribute('aria-label')) {
-    value = element.getAttribute('aria-label')!
-  } else if (element.textContent) {
-    value = element.textContent
+type TextEquivalentOptions = {
+  allowHidden: boolean
+}
+
+const defaultOptions: TextEquivalentOptions = {
+  allowHidden: false,
+}
+
+/**
+ * Simplified version of the algorithm to compute the text equivalent of an
+ * element.
+ *
+ * @see https://www.w3.org/TR/accname-1.2/#computation-steps
+ */
+function computeTextEquivalent(
```

Comment:
This seems like a good start - I think we can refine this as we identify different usecases (alongside stories/tests).